poc/Cornelius.tbasic

```diff
@@ -0,0 +1,755 @@
+﻿'---Script created on 04-25-2020 20:39:06 by
+uses "console"
+Uses "TBGL"
+Uses "UI"
+
+#include "gameZoneDefinition.tbasicu"
+#Include "https://github.com/DirectuX/logger/raw/master/LOG.tbasicu"
+
+Global logger as tLogThis
+logger.defaultTarget = %LT_ToConsole
+Console_SetScreenBufferSize(120,300)
+
+  Type tRiver
+      minSectorSurface as dWord
+      minHalfWidth as byte
+      maxHalfWidth as byte
+      banksWidth as byte
+      seed as Number
+      meandering as single
+      meander as Single
+End Type
+
+  Type tXY
+      x as DWORD
+      y as DWORD
+End Type
+
+' To store flow direction in waypoints
+  begin Const '
+  %NORTH         = 1
+  %SOUTH         = 2
+  %EAST          = 4
+  %WEST         = 8
+  end const
+
+Function TBMain()
+
+  logger.defaultTitle = Function_Name
+  logger.wInfo("Logger loaded")
+
+  call parametersSetting ' Sets initial variables
+  Call initDisplay ' Create the windows for map display
+  Call initWorldMap ' Build straight river
+  Call stirWorldMap
+  CAll sleepDisplay ' Final loop to keep displaying map
+
+End Function
+
+function parametersSetting()
+
+  logger.defaultTitle = Function_Name
+
+  global displayWindowSize as tXY
+  global hWnd AS DWORD
+
+  global myRiver as tRiver
+  Global actualRiverStatus as tRiver
+
+  myRiver.banksWidth = 3
+  myRiver.minHalfWidth = 1
+  myRiver.maxHalfWidth = 4 ' typical = 3 to 12
+  myRiver.minSectorSurface = 100 * myRiver.maxHalfWidth
+  myRiver.meandering = 0.15 ' how is the meander distribution ( far-off and nearby disclosed % ) typical = 0.15
+  myRiver.meander = 0.9 ' how max long is meander ( is a % of stretch length ) typical = 0.9
+  myRiver.seed = 23031 ' try different values ' 23031
+
+  Global maxLarge as Byte = 1 + myRiver.maxHalfWidth * 2 ' max river width, this is also the lakes's minimum side
+  global mapDimensions as tXY
+
+  for i as byte = myRiver.minHalfWidth to myRiver.maxHalfWidth
+    mapDimensions.y += Fix(myRiver.minSectorSurface/i^0.5)
+  next i
+
+  mapDimensions.x = fix(mapDimensions.y / 4)
+
+  global actualPosition as tXY
+
+'[todo] add a calm pool there (minimum lake size should be ok)
+'Start (source) of the river
+  actualPosition.y = 0
+  actualPosition.x = fix(mapDimensions.x / 2)
+
+  global map as tGameZoneDefinition()
+
+  logger.write(%LT_ToMsgBox + %LT_ToConsole, mapDimensions.x & " x " & mapDimensions.y, "mapDimensions", %TAG_INFO)
+
+End Function
+
+function initDisplay()
+
+  DESKTOP GET CLIENT TO displayWindowSize.x, displayWindowSize.y
+
+  hWnd = tbgl_CreateWindowEx("press ESC to quit", min(mapDimensions.x, displayWindowSize.x / 2), min(mapDimensions.y, displayWindowSize.y), 32, %tbgl_WS_WINDOWED or %tbgl_WS_CLOSEBOX, -15 + displayWindowSize.x - min(mapDimensions.x, displayWindowSize.x / 2), 0)
```

Comment:
Alternative would be to create window "somehow" and adjust the size via DIALOG functions:

```
  DESKTOP GET CLIENT TO displayWindowSize.x, displayWindowSize.y
  
  dWord hWnd = tbgl_createWindowEx("TBGL sample script - press ESC to quit", 512, 512, 32, %TBGL_WS_WINDOWED | %TBGL_WS_CLOSEBOX | %TBGL_WS_DONTKEEPASPECTRATIO) ' %TBGL_WS_DONTKEEPASPECTRATIO is important here
  tbgl_showWindow

  dialog set Size hWnd, displayWindowSize.x, displayWindowSize.y  ' Set size
  dialog set Loc hWnd, 1, 1                                       ' Set position
```

Reply:
I continue with the first suggestion, at least for now.

Reply:
Wished to say that pixel_perfect is perfect. I think I didn't use it well as same calculations on the same picture displays differently at various places. I'm publishing the changes, then I'll try the alternative.

Reply:
see at the bottom, the line should be 3px wide. Same formula (%NORTH widen) for both, but one horizontal line is narrower in display.
![image](https://user-images.githubusercontent.com/17156661/81795517-7eef9c80-950c-11ea-92a7-7556a702a9d0.png)

Reply:
Hi Sebastian,

I can confirm something strange is happening here for sure, investigating...

Reply:
@DirectuX - either we can hunt bugs in different GPU drivers / my implementation of pixel perfect, or we could... use the right tool for the job.

For the game, we will use TBGL for sure, it is handy for fast geometry drawing. However - for algorithm debugging? Canvas_Window is the best. I proposed the change here:
https://github.com/petrSchreiber/BetweenTheBubbles-tb/pull/28/files

Reply:
thanks for the help !
just checked the pixels and the formula are correct !
Still the randomization to check. will be easier now, I hope.